### DISCUSSION TITLE
Unable to build minting-dapp because of dependency from the smart-contract folder

### DISCUSSION THREAD

**Discussion by Solonnikov** [hashlips-lab/nft-erc721-collection#221] [Q&A] [answered]
Guys, when I try to build minting-dapp FE on Netlify or Vercel i got an error - Module build failed: Module not found: "./src/scripts/react/Dapp.tsx" contains a reference to the file "../../../../smart-contract/artifacts/contracts".

It looks for dependency from the sibling folder. Of course, it is okay when I build it locally, but not on the repo.

Any advice? Thanks

**Comment by liarco** [comment, marked as answer]
Hi @Solonnikov,
That interdependency is there in order to avoid duplicating configuration and contract artifacts between the two projects (that's also why we have a single repository for both the smart contract and the DAPP).
While this improves the developer experience, I'm aware that it causes issues with some automated deployment services and that's why I'm working on an improvement for the next version (it will be out soon, with a video tutorial coming next week).

There is no stable version including the changes since I'm still working on it, but the dedicated PR is #208.

**Reply by Solonnikov** [reply]
Thank you, Marco. Will take a look. And thanks for this incredible work :)
